ui/app/templates/rules.html

```diff
@@ -37,57 +43,110 @@
+        </button>
+        <button ng-show="!rule_id && show_sc && rule.scheduled_change.change_type === 'insert'" class="btn btn-info btn-xs"
+                ng-click="openEditScheduledRuleChangeModal(rule)" title="{{ humanizeDate(rule.scheduled_change.when) }}">
+            Pending Creation <span ng-bind-html="formatMoment(rule.scheduled_change.when)"></span>
+        </button>
+        <button ng-show="!rule_id && show_sc && rule.scheduled_change.change_type === 'update'" class="btn btn-warning btn-xs"
+                ng-click="openEditScheduledRuleChangeModal(rule)" title="{{ humanizeDate(rule.scheduled_change.when) }}">
+            Pending Update <span ng-bind-html="formatMoment(rule.scheduled_change.when)"></span>
+        </button>
+        <button ng-show="!rule_id && show_sc && rule.scheduled_change.change_type === 'delete'" class="btn btn-danger btn-xs"
+                ng-click="openEditScheduledRuleChangeModal(rule)" title="{{ humanizeDate(rule.scheduled_change.when) }}">
+            Pending Delete <span ng-bind-html="formatMoment(rule.scheduled_change.when)"></span>
+        </button>
+        <button ng-show="!rule_id && rule.scheduled_change === null" class="btn btn-warning btn-xs"
+                ng-click="openNewScheduledRuleChangeModal(rule)">
+            Schedule an Update
+        </button>
+        <button ng-show="!rule_id && rule.scheduled_change === null" class="btn btn-danger btn-xs"
+                ng-click="openNewScheduledDeleteModal(rule)">
+            Schedule for Deletion
+        </button>
+        <button ng-show="!rule_id && rule.scheduled_change !== null && rule.scheduled_change.change_type !== 'insert'"
+                class="btn btn-default btn-xs" ng-click="openUpdateModal(rule)">
+            Update
+        </button>
+        <button ng-show="!rule_id && rule.scheduled_change !== null && rule.scheduled_change.change_type !== 'insert'"
+                class="btn btn-default btn-xs" ng-click="openDeleteModal(rule)">
+            Delete
+        </button>
+        <button ng-show="!rule_id && rule.scheduled_change !== null && rule.scheduled_change.change_type !== 'insert'"
+                class="btn btn-default btn-xs" ng-click="openDuplicateModal(rule)">
+            Duplicate
+        </button>
+        <a ng-show="!rule_id && rule.scheduled_change !== null && rule.scheduled_change.change_type !== 'insert'"
+           class="btn btn-default btn-xs" ng-href="/rules/{{ rule.rule_id }}">
+            History
+        </a>
       </div>
-
-      <i title="Product" ng-show="!rule.product">no product</i>
-      <b title="Product" ng-show="rule.product" ng-bind-html="highlightSearch(rule.product, 'product')"></b>
+      <i title="Product" ng-show="!rule.product && !rule.scheduled_change.product">no product</i>
+      <b title="Product">
+        <span ng-show="rule_id || !show_sc || rule.scheduled_change === null || rule.product === rule.scheduled_change.product">{{ rule.product }}</span>
```

Comment:
I think we can factorize similar conditions thanks to `rulefield` 😃

Reply:
Sorry, can you expand on this?

Reply:
Yeah, sorry about this.

Basically, the condition at this line, is almost identical to the one in `rulefield`. Moreover, we want similar behaviors on strikethroughs, per Nick's screenshot. So my point was: because the logic is the same, we can use rulefield here. What do you think?

Reply:
We might be able to move it into a helper function and pass show_sc and rule to it...I'll investigate that.